src/library/scala/collection/mutable/ArraySeq.scala

```diff
@@ -269,4 +272,25 @@ object ArraySeq extends StrictOptimizedClassTagSeqFactory[ArraySeq] { self =>
   // This prevents it from serializing it in the first place:
   private[this] def writeObject(out: ObjectOutputStream): Unit = ()
   private[this] def readObject(in: ObjectInputStream): Unit = ()
+
+  private[collection] def deep[T](ms: ArraySeq[T]): scala.collection.IndexedSeq[Any] = {
+    new scala.collection.AbstractSeq[Any] with scala.collection.IndexedSeq[Any] {
+      def length = ms.length
+      def apply(idx: Int): Any = ms.apply(idx) match {
+        case null              => null
+        case x: Array[AnyRef]  => deep((new mutable.ArraySeq.ofRef[AnyRef](x)).asInstanceOf[mutable.ArraySeq[AnyRef]] )
+        case x: Array[Int]     => deep((new mutable.ArraySeq.ofInt(x)        ).asInstanceOf[mutable.ArraySeq[Int]]    )
+        case x: Array[Double]  => deep((new mutable.ArraySeq.ofDouble(x)     ).asInstanceOf[mutable.ArraySeq[Double]] )
+        case x: Array[Long]    => deep((new mutable.ArraySeq.ofLong(x)       ).asInstanceOf[mutable.ArraySeq[Long]]   )
+        case x: Array[Float]   => deep((new mutable.ArraySeq.ofFloat(x)      ).asInstanceOf[mutable.ArraySeq[Float]]  )
+        case x: Array[Char]    => deep((new mutable.ArraySeq.ofChar(x)       ).asInstanceOf[mutable.ArraySeq[Char]]   )
+        case x: Array[Byte]    => deep((new mutable.ArraySeq.ofByte(x)       ).asInstanceOf[mutable.ArraySeq[Byte]]   )
+        case x: Array[Short]   => deep((new mutable.ArraySeq.ofShort(x)      ).asInstanceOf[mutable.ArraySeq[Short]]  )
+        case x: Array[Boolean] => deep((new mutable.ArraySeq.ofBoolean(x)    ).asInstanceOf[mutable.ArraySeq[Boolean]])
+        case x: Array[Unit]    => deep((new mutable.ArraySeq.ofUnit(x)       ).asInstanceOf[mutable.ArraySeq[Unit]]   )
+        case x => x
+      }
```

Comment:
The current implementation in 2.12.x seems simpler: https://github.com/scala/scala/blob/2.12.x/src/library/scala/collection/mutable/ArrayLike.scala#L41-L44

Is there a reason to have all these `case`s?